packages/react/src/components/DetailsList/DetailsRow.base.tsx

```diff
@@ -327,7 +327,7 @@ export class DetailsRowBase extends React.Component<IDetailsRowBaseProps, IDetai
         className={this._classNames.root}
         data-selection-index={itemIndex}
         data-selection-touch-invoke={true}
-        data-selection-disabled={disabled || undefined}
+        data-selection-disabled={(this.props as any)['data-selection-disabled'] ?? disabled ?? undefined}
```

Comment:
This logic for this prop is getting more complex. Perhaps we should have a test or two?